src/StoreKeeper/WooCommerce/B2C/Tools/Media.php

```diff
@@ -210,4 +315,102 @@ public static function tryDownloadFile($url)
 
         return $response;
     }
+
+    /**
+     * Removes the prepended WordPress upload path in the URL if attachment comes from CDN.
+     *
+     * @see wp_get_attachment_url
+     */
+    public function getAttachmentUrl($attachmentUrl, $attachmentId): string
+    {
+        if ($this->isAttachmentCdn($attachmentId)) {
+            $attachmentUrl = self::cleanAttachmentUrl($attachmentUrl);
+        }
+
+        return urldecode($attachmentUrl);
+    }
+
+    /**
+     * Changes the image source to CDN variant if a size matches.
+     * Removes the prepended WordPress upload path in the URL if attachment comes from CDN otherwise.
+     * Returns original file path otherwise.
+     *
+     * @see wp_get_attachment_image_src
+     */
+    public function getAttachmentImageSource($attachmentImage, $attachmentId, $attachmentSize, $isAttachmentIcon)
```

Comment:
it's good to add some unit tests for whole decode/encode of the images

Reply:
Added test for getting of attachment url, source, and source set.